utilities/architecture.py

```
@@ -1,37 +1,36 @@
-
-    arch = X86_64 if arch == AMD_64 else arch
+        return {node.labels[KUBERNETES_ARCH_LABEL] for node in nodes}
 
-    if arch not in (X86_64, ARM_64, S390X):
-        raise ValueError(f"{arch} architecture in not supported")
-
-    return arch
+    if not arch:
+        raise UnsupportedCPUArchitectureError(
+            "Cluster architecture could not be determined (no nodes found and env var unset)."
+        )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)